backend/src/controllers/formContoller.ts

```diff
@@ -1,22 +1,36 @@
-import { Response } from "express";
-import { createFormValidation } from "../validations/formValidation";
+import { Request, Response } from "express";
+import {
+  createFormValidation,
+  editFormValidation,
+} from "../validations/formValidation";
 import Form from "../models/Form";
+import Joi from "joi";
+import { CreateFormType, Search, FormType } from "../utils/types";
+import Question from "../models/Question";
 
-export const createForm = async (req: any, res: Response) => {
+export const createForm = async (req: Request, res: Response) => {
   try {
-    const validationResult = await createFormValidation.validateAsync(req.body);
-    const { title, description } = validationResult;
+    const validationResult: Joi.ValidationResult<CreateFormType> =
+      createFormValidation.validate(req.body);
+    if (validationResult.error) {
+      return res.status(400).json({ message: validationResult.error.message });
+    }
+
+    const { title, description }: CreateFormType = req.body;
     const createdForm = await Form.create({ title, description });
     return res.status(201).json(createdForm);
-  } catch (error: any) {
+  } catch (error) {
     return res.status(400).json({ error });
   }
 };
 
-export const getForms = async (req: any, res: Response) => {
+export const getForms = async (
+  req: Request<object, object, object, Search>,
+  res: Response,
+) => {
   try {
     const searchString = req.query.searchString || "";
-    const forms = await Form.aggregate([
+    const forms: FormType[] = await Form.aggregate([
```

Comment:
There is no separation of layers here. Controller should not access directly the repository/domain layer that should be done in a Service Layer.Smth along these lines controller -> service -> repository